copy_test.go

```diff
@@ -352,3 +360,97 @@ func TestCopyGraph_PartialCopy(t *testing.T) {
 		t.Errorf("count(dst.Exists()) = %v, want %v", got, want)
 	}
 }
+
+func testService() *httptest.Server {
+	const exampleRepoName = "example"
+	const exampleDigest = "sha256:aafc6b9fa2094cbfb97eca0355105b9e8f5dfa1a4b3dbe9375a30b836f6db5ec"
+	const exampleTag = "latest"
+	const exampleBlob = "Example blob content"
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		p := r.URL.Path
+		m := r.Method
+		switch {
+		case strings.HasSuffix(p, "/blobs/uploads/") && m == "GET":
+			w.WriteHeader(http.StatusCreated)
+		case (strings.HasSuffix(p, "/manifests/latest") || strings.HasSuffix(p, fmt.Sprintf("/manifests/%s", exampleDigest))) && m == "HEAD":
+			w.Header().Set("Content-Type", ocispec.MediaTypeImageLayer)
+			w.Header().Set("Docker-Content-Digest", exampleDigest)
+			w.Header().Set("Content-Length", strconv.Itoa(len([]byte(exampleBlob))))
+		case strings.HasSuffix(p, fmt.Sprintf("/manifests/%s", exampleDigest)) && m == "GET":
+			w.Header().Set("Content-Type", ocispec.MediaTypeImageLayer)
+			w.Header().Set("Docker-Content-Digest", exampleDigest)
+			w.Header().Set("Content-Length", strconv.Itoa(len([]byte(exampleBlob))))
+			w.Write([]byte(exampleBlob))
+		case strings.HasSuffix(p, fmt.Sprintf("/blobs/%s", exampleDigest)) && (m == "GET" || m == "HEAD"):
+			w.Header().Set("Content-Type", ocispec.MediaTypeImageLayer)
+			w.Header().Set("Docker-Content-Digest", exampleDigest)
+			w.Header().Set("Content-Length", strconv.Itoa(len([]byte(exampleBlob))))
+			w.Write([]byte(exampleBlob))
+		case strings.HasSuffix(p, fmt.Sprintf("/manifests/%s", exampleTag)) && m == "PUT":
+			w.WriteHeader(http.StatusCreated)
+		}
+
+	}))
+}
+
+func getTestRepository(ctx context.Context, repositoryName string) (ts *httptest.Server, repo registry.Repository, err error) {
+	ts = testService()
+
+	uri, err := url.Parse(ts.URL)
+	if err != nil {
+		// handle it
+		return
+	}
+	reg, err := remote.NewRegistry(uri.Host)
+	if err != nil {
+		// handle it
+		return
+	}
+	reg.RepositoryOptions.PlainHTTP = true
+	repo, err = reg.Repository(ctx, repositoryName)
+	return ts, repo, err
+}
+
+func ExampleCopy() {
+	ctx := context.Background()
+	ts1, sourceRepo, err1 := getTestRepository(ctx, "source")
+	ts2, targetRepo, err2 := getTestRepository(ctx, "target")
+	if err1 != nil || err2 != nil {
+		// handle it
+		return
+	}
+	defer ts1.Close()
```

Comment:
1. Remove ts1 and ts1 defer. Does not pertain to using this function - Is there any issue if we dont' defer it. I think the test framework will take it down, but you can confirm. 
2. use `reg.Respository(ctx. "source")` instead hidden function - See the generated docs and it will be hard to follow and users will wonder what is getRepository doing. 

I think if you use Testing Main then you can just have a var that has the `reg` testregistry.

Reply:
1. If it's not deferred, then the test http server will be closed before receving mocked request, I have confirmed.
2. I will refactor the implementation like registry examples, remove the getTestRepository function and other stuff. Making this PR as draft before the refactoring is pushed.